### PR TITLE
W&B Logger updated docstring and parameters

### LINKED ISSUE #511: Expand input parameters to the W&BLogger class
Currently, WandBLogger accepts a `init_kwargs` as input parameter. I think we should avoid a dictionary if possible and prefer explicit parameters passed to the constructor. Most of them should be made optional. This can be implemented directly in #483 .
For example, by default the `project` and `run` parameters could be set to `avalanche` and `test`, respectively.

### PR DESCRIPTION
**Pull request by digantamisra98**
This PR fixes issue #511.

### CODE REVIEW

File: avalanche/logging/wandb_logger.py [hunk 1]
```diff
@@ -28,17 +27,19 @@ class WandBLogger(StrategyLogger):
     logged to a dedicated Weights & Biases project dashboard.
     """
 
-    def __init__(self, init_kwargs: dict, interactive: bool = False):
+    def __init__(self, project_name: str, run_name: str, params: dict):
```
Comment (AndreaCossu) [inline avalanche/logging/wandb_logger.py:30]: Can you make `params` optional by setting it by default to `{}`?

File: avalanche/logging/wandb_logger.py [hunk 2]
```diff
@@ -28,17 +27,19 @@ class WandBLogger(StrategyLogger):
...
         """
         Creates an instance of the `WandBLogger`.
 
-        :param init_kwargs: All arguments for wandb.init() function call.:
+        :param params: All arguments for wandb.init() function call.:
```
Comment (AndreaCossu) [inline avalanche/logging/wandb_logger.py:36]: If maintaining here the list of arguments is challenging, can you put a link to the W&B documentation for `init` call?

Comment (digantamisra98) [reply avalanche/logging/wandb_logger.py:36]: @AndreaCossu if I understood correctly you want me to put a URL to the W&B init params docs in the docstrings?

File: avalanche/logging/wandb_logger.py (hunk 1 above)
Comment (digantamisra98) [reply avalanche/logging/wandb_logger.py:30]: Sure, yes

File: avalanche/logging/wandb_logger.py (hunk 2 above)
Comment (AndreaCossu) [reply avalanche/logging/wandb_logger.py:36]: exactly :smile:

File: avalanche/logging/wandb_logger.py [hunk 3]
```diff
@@ -49,6 +55,11 @@ def import_wandb(self):
                 'Please run "pip install wandb" to install wandb')
         self.wandb = wandb
 
+    def args_parse(self):
+        self.init_kwargs = {"project": self.project_name, "name": self.run_name}
+        if self.params:
+            self.init_kwargs |= self.params
```
Comment (AndreaCossu) [inline avalanche/logging/wandb_logger.py:61]: Maybe here you want to merge the two dictionaries? I don't think the `|=` operation is valid on two dicts.

Comment (digantamisra98) [reply avalanche/logging/wandb_logger.py]: Oh right, fixed the merge. My bad

**Comment by AndreaCossu** [comment]
Great, thanks! I'll merge the PR as soon as tests are completed :smile:

**Comment by digantamisra98** [comment]
Thanks!!

**Pull request merged**